include/deal.II/lac/precondition.h

```diff
@@ -569,6 +569,28 @@ namespace internal
     template <typename T, typename VectorType>
     const bool has_step<T, VectorType>::value;
 
+    template <typename T, typename VectorType>
+    struct has_step_omega
+    {
+    private:
+      static bool
+      detect(...);
+
+      template <typename U>
+      static decltype(
+        std::declval<U const>().step(std::declval<VectorType &>(),
+                                     std::declval<const VectorType &>(),
+                                     std::declval<const double>()))
+      detect(const U &);
+
+    public:
+      static const bool value =
+        !std::is_same<bool, decltype(detect(std::declval<T>()))>::value;
```

Comment:
Let's turn this around to make it read like "if this complicated type is 'bool'":
```suggestion
        !std::is_same<decltype(detect(std::declval<T>())),bool>::value;
```